lib/caching/NonceCache.ts

```diff
@@ -150,12 +150,20 @@ export class NonceCache {
   // We wait for a maximum of MAX_NONCE_LOCK_TIME_MS until a nonce lock is released
   private async waitForLockAndGetNonceInfo(
     signerAddress: string,
-    nonceType: string,
+    nonceType: NonceType,
     requestId: string
   ): Promise<CachedNonceInfo> {
     log.debug(new Date(), ` ${requestId} - Waiting for nonce to be unlocked. Max wait: ${MAX_NONCE_LOCK_TIME_MS}ms`);
-    for (let i = 0; i < Math.ceil(MAX_NONCE_LOCK_TIME_MS / NONCE_LOCK_POLL_INTERVAL_MS); i++) {
+    const maxIteration = Math.ceil(MAX_NONCE_LOCK_TIME_MS / NONCE_LOCK_POLL_INTERVAL_MS) + 1;
```

Comment:
do we need the `+1` here?

Reply:
We don't, I have removed it. I was leftover from a different algorithm.